File: prototype/php/src/app/Domain/Listings/ListingEventTypeTest.php
<?php

declare(strict_types=1);

namespace App\Domain\Listings;

it('names the storefront interactions the seller reports on', function (): void {
    expect(array_column(ListingEventType::cases(), 'value'))
        ->toBe(['view', 'favorite', 'unfavorite', 'cart_add']);
});

File: prototype/rails/src/app/domain/listings/listing_event_type_test.rb
# Runs without Rails: ruby -Iapp app/domain/listings/listing_event_type_test.rb
require "minitest/autorun"
require_relative "listing_event_type"

module Domain
  module Listings
    class ListingEventTypeTest < Minitest::Test
      def test_all_names_every_event_a_listing_records
        assert_equal %w[view favorite unfavorite cart_add], ListingEventType::ALL
      end
    end
  end
end
